internal/controller/controllers/agentserviceconfig_controller_test.go

```diff
@@ -104,13 +106,145 @@ var _ = Describe("agentserviceconfig_controller reconcile", func() {
 
 	BeforeEach(func() {
 		asc = newASCDefault()
-		ascr = newTestReconciler(asc, ingressCM, route, imageRoute)
 	})
 
 	It("reconcile should succeed", func() {
+		agentinstalladmissionDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "agentinstalladmission",
+				Namespace: "assisted-installer",
+			},
+		}
+		imageServiceDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "assisted-image-service",
+				Namespace: "assisted-installer",
+			},
+		}
+		assistedServiceDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "assisted-service",
+				Namespace: "assisted-installer",
+			},
+		}
+		ascr = newTestReconciler(asc, ingressCM, route, imageRoute, agentinstalladmissionDeployment, imageServiceDeployment, assistedServiceDeployment)
+		result, err := ascr.Reconcile(ctx, newAgentServiceConfigRequest(asc))
+		Expect(err).To(Succeed())
+		Expect(result).To(Equal(ctrl.Result{}))
+	})
+
+	It("should set `DeploymentsHealthy` condition to `False` on AgentServiceConfig when a deployment is not Available", func() {
+		agentinstalladmissionDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "agentinstalladmission",
+				Namespace: "assisted-installer",
+			},
+			Status: appsv1.DeploymentStatus{
+				Conditions: []appsv1.DeploymentCondition{
+					{
+						Type:   appsv1.DeploymentAvailable,
+						Status: corev1.ConditionTrue,
+					},
+				},
+			},
+		}
+		imageServiceDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "assisted-image-service",
+				Namespace: "assisted-installer",
+			},
+			Status: appsv1.DeploymentStatus{
+				Conditions: []appsv1.DeploymentCondition{
+					{
+						Type:   appsv1.DeploymentAvailable,
+						Status: corev1.ConditionTrue,
+					},
+				},
+			},
+		}
+		assistedServiceDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "assisted-service",
+				Namespace: "assisted-installer",
+			},
+			Status: appsv1.DeploymentStatus{
+				Conditions: []appsv1.DeploymentCondition{
+					{
+						Type:   appsv1.DeploymentAvailable,
+						Status: corev1.ConditionFalse,
+					},
+				},
+			},
+		}
+		ascr = newTestReconciler(asc, ingressCM, route, imageRoute, agentinstalladmissionDeployment, imageServiceDeployment, assistedServiceDeployment)
 		result, err := ascr.Reconcile(ctx, newAgentServiceConfigRequest(asc))
+
+		Expect(err).NotTo(Succeed())
+		Expect(result).NotTo(Equal(ctrl.Result{}))
+
+		instance := &aiv1beta1.AgentServiceConfig{}
+		err = ascr.Get(ctx, types.NamespacedName{Name: "agent"}, instance)
+
+		Expect(err).To(BeNil())
+		Expect(instance.Status.Conditions[1].Type, aiv1beta1.ConditionDeploymentsHealthy)
+		Expect(instance.Status.Conditions[1].Status, corev1.ConditionFalse)
```

Comment:
Is this guaranteed to always be the condition at index 1?
This could lead to a flaky test if the order isn't deterministic.

Reply:
Good catch. Updated it.